hexkit/inject.py

```diff
@@ -29,82 +29,87 @@
 import inspect
 from collections.abc import AsyncIterator
 from contextlib import AbstractAsyncContextManager
-from typing import Any, Callable, Generic, Optional, TypeVar
+from typing import Any, Callable, Generic, Optional, TypeVar, Union
 
 import dependency_injector.containers
 import dependency_injector.providers
 from pydantic import BaseSettings
 
-from hexkit.custom_types import ContextConstructable
+from hexkit.custom_types import AsyncConstructable, AsyncContextConstructable
 
 __all__ = [
     "get_constructor",
     "ContainerBase",
     "NotConstructableError",
-    "ContextConstructor",
+    "AsyncConstructor",
     "AsyncInitShutdownError",
     "Configurator",
 ]
 
 
 class NotConstructableError(TypeError):
-    """Thrown when a ContextConstructable expected but not obtained."""
+    """Thrown when a AsyncContextConstructable expected but not obtained."""
 
 
 class AsyncInitShutdownError(TypeError):
     """Thrown when a container has sync `init_resource` or `shutdown_resource` methods
     but coroutines are needed."""
 
 
-def assert_context_constructable(constructable: type[ContextConstructable]):
+def assert_async_constructable(constructable: type[AsyncContextConstructable]):
     """
     Make sure that the provided object has a callable attribute `construct`.
     If this check passes, it can be seen as a strong indication that the provided object
-    is compliant with our definition of a ContextConstructable. However, it does not
-    check whether `construct` really returns an async context manager.
+    is compliant with our definition of a Async(Context)Constructable. However, it does
+    not check whether `construct` really returns an awaitable or an async context
+    manager.
     """
```

Comment:
Maybe use `iscoroutinefunction` here?

Reply:
That will not match:
````
@asynccontextmanager
async def construct(cls, *args: Any, **kwargs: Any): ...
```

Since construct returns an async context manager and not an awaitable.